src/components/UIUC-Components/EditCourseCard.tsx

```diff
@@ -168,6 +168,18 @@ const EditCourseCard = ({
     setApiKey(courseMetadata?.openai_api_key as string)
   }, [courseMetadata])
 
+  const validateKey = async (key: string) => {
```

Comment:
Quick check, is this safe? 

Are we sending an un-encrypted key over the internet?

Reply:
The key is added to request body on the client and sent to server. Ideally https protocol encrypts all body data, primarily the reason I used `POST`, similar to how it is handled in `models.ts`. It can be insecure only if someone intercepts the HTTP call from another client and is able to decrypt the request body, otherwise the data is secured on client-server connection.

Reply:
Ok I agree it is secure. 

1. It's using HTTPS, so the data is encrypted in flight. This is all that's required for security.

Furthermore, our CORS policy prevents any other domains from sending our backend weird data, which is just a bonus so someone's course metadata is not corrupted.